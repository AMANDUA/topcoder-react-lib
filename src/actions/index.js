/**
 * Export actions.
 */
import authActions from './auth';
import smpActions from './smp';
import statsActions from './stats';
import termsActions from './terms';
import directActions from './direct';
import groupActions from './groups';
import errorActions from './errors';
import challengeActions from './challenge';
import profileActions from './profile';
import memberActions from './members';
import memberTaskActions from './member-tasks';
import reviewOpportunityActions from './reviewOpportunity';

export const actions = {
  auth: authActions.auth,
  smp: smpActions.smp,
  challenge: challengeActions.challenge,
  terms: termsActions.terms,
  stats: statsActions.stats,
  direct: directActions.direct,
  groups: groupActions.groups,
  errors: errorActions.errors,
  profile: profileActions.profile,
  members: memberActions.members,
  memberTasks: memberTaskActions.memberTasks,
  reviewOpportunity: reviewOpportunityActions.reviewOpportunity,
};

export default undefined;
