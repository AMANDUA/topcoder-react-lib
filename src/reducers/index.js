/**
 * Export reducers.
 */
import * as auth from './auth';
import * as stats from './stats';
import * as terms from './terms';
import * as direct from './direct';
import * as groups from './groups';
import * as errors from './errors';
import * as challenge from './challenge';
import * as profile from './profile';
import * as members from './members';
import * as memberTasks from './member-tasks';
import * as reviewOpportunity from './reviewOpportunity';
import * as mySubmissionsManagement from './my-submissions-management';

export const reducers = {
  auth,
  stats,
  terms,
  direct,
  groups,
  errors,
  challenge,
  profile,
  members,
  memberTasks,
  reviewOpportunity,
  mySubmissionsManagement,
};

export default undefined;
