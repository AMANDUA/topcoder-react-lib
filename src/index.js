/**
 * Export the lib.
 */
import reducers, { factory as reducerFactory } from './reducers';

export { reducerFactory };

export { reducers };

export { actions } from './actions';

export { services } from './services';

export { challenges, logger, errors, tc, time, mock } from './utils';
