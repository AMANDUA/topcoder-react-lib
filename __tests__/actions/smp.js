import { actions } from '../../src';

test('smp.deleteSubmissionDone', () => {
  expect(actions.smp.deleteSubmissionDone('Token V3', 'submissionId'))
    .toMatchSnapshot();
});

describe('smp.downloadSubmission', () => {
  test('does not throw', () => {
    expect(() => actions.smp.downloadSubmission({}, 'design', '12345')).not.toThrow();
  });
});
