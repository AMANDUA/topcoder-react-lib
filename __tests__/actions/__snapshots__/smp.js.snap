// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smp.deleteSubmissionDone 1`] = `
Object {
  "payload": Promise {},
  "type": "SMP/DELETE_SUBMISSION_DONE",
}
`;
